jgrapht-core/src/test/java/org/jgrapht/GraphsTest.java

```diff
@@ -455,6 +458,1255 @@ public void vertexHasParents_A()
+
+        String a = "A";
+        String b = "B";
+
+        DefaultEdge edge = Graphs.<String, DefaultEdge>addEdgeWithVertices(graph, a, b);
+
+        Assert.assertNotEquals(null, edge);
+        Assert.assertTrue(graph.getEdgeSource(edge).equals(a));
+        Assert.assertTrue(graph.getEdgeTarget(edge).equals(b));
+    }
+
+    /*
+     *  Tests for public static <V,
+     *      E> boolean addEdgeWithVertices(Graph<V, E> targetGraph, Graph<V, E> sourceGraph, E edge)
+     */
+    @Test
+    public void addEdgeWithVertices2ExistingVerticesTest() {
+        SimpleGraph<String, DefaultEdge> graphSource = new SimpleGraph<>(DefaultEdge.class);
+        SimpleGraph<String, DefaultEdge> graphTarget = new SimpleGraph<>(DefaultEdge.class);
+
+        String a = "A";
+        String b = "B";
+
+        graphSource.addVertex(a);
+        graphSource.addVertex(b);
+
+        graphTarget.addVertex(a);
+        graphTarget.addVertex(b);
+
+        DefaultEdge edgeSource = Graphs.<String, DefaultEdge>addEdgeWithVertices(graphSource, a, b);
+
+        Boolean result = Graphs.<String, DefaultEdge>addEdgeWithVertices(graphTarget, graphSource, edgeSource);
+
+        DefaultEdge edgeTarget = graphTarget.getEdge(a, b);
+        edgeSource = graphSource.getEdge(a, b);
+
+        Assert.assertTrue(result);
+        Assert.assertTrue(graphTarget.getEdgeSource(edgeTarget).equals(a));
+        Assert.assertTrue(graphTarget.getEdgeTarget(edgeTarget).equals(b));
+        Assert.assertTrue(graphSource.getEdgeSource(edgeSource).equals(a));
+        Assert.assertTrue(graphSource.getEdgeTarget(edgeSource).equals(b));
+    }
+
+    @Test
+    public void addEdgeWithVertices2NewVerticesTest() {
+        SimpleGraph<String, DefaultEdge> graphSource = new SimpleGraph<>(DefaultEdge.class);
+        SimpleGraph<String, DefaultEdge> graphTarget = new SimpleGraph<>(DefaultEdge.class);
+
+        String a = "A";
+        String b = "B";
+
+        graphSource.addVertex(a);
+        graphSource.addVertex(b);
+
+        DefaultEdge edgeSource = Graphs.<String, DefaultEdge>addEdgeWithVertices(graphSource, a, b);
+
+        Boolean result = Graphs.<String, DefaultEdge>addEdgeWithVertices(graphTarget, graphSource, edgeSource);
+
+        DefaultEdge edgeTarget = graphTarget.getEdge(a, b);
+        edgeSource = graphSource.getEdge(a, b);
+
+        Assert.assertTrue(result);
+        Assert.assertTrue(graphTarget.getEdgeSource(edgeTarget).equals(a));
+        Assert.assertTrue(graphTarget.getEdgeTarget(edgeTarget).equals(b));
+        Assert.assertTrue(graphSource.getEdgeSource(edgeSource).equals(a));
+        Assert.assertTrue(graphSource.getEdgeTarget(edgeSource).equals(b));
+    }
+
+    /*
+     *  Tests for public static <V,
+     *      E> E addEdgeWithVertices(Graph<V, E> g, V sourceVertex, V targetVertex, double weight)
+     */
+    @Test
+    public void addEdgeWithVertices3ExistingVerticesTest() {
```

Comment:
What's the purpose of this test compared to the previous test. Seems a duplicate.